lisa/wlgen/workload.py

```diff
@@ -80,7 +82,18 @@ def __init__(self, target, name, res_dir=None):
         self.name = name
         self.command = None
         self.output = ""
-        self.run_dir = self.target.working_directory
+
+        wlgen_dir = self.target.path.join(target.working_directory,
+                                          "lisa", "wlgen")
+        self.target.execute("mkdir -p {}/lisa/wlgen".format(quote(wlgen_dir)))
+
+        timestamp = datetime.now().strftime('%Y%m%d_%H%M%S')
+        self.run_dir = target.execute("mktemp -d -p {} {}_{}_XXXXXX"
```

Comment:
individual parameters of the command need to be built "separately" and then quoted as independent entities using `shlex.quote()`:
```
         temp_template = "{}_{}_XXXXXX".format(self.name, timestamp)
         self.run_dir = target.execute("mktemp -d -p {} ".format(
                                      .format(quote(wlgen_dir), quote(temp_template)).strip()
```